stubs/sublime.pyi

```diff
@@ -1388,7 +1388,7 @@ class Selection(Reversible):
         """
         ...
 
-    def __eq__(self, rhs: Selection | None) -> bool:
+    def __eq__(self, rhs: object) -> bool:
```

Comment:
```
stubs/sublime.pyi:1391: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"  [override]
stubs/sublime.pyi:1391: note: This violates the Liskov substitution principle
stubs/sublime.pyi:1391: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
stubs/sublime.pyi:1391: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
stubs/sublime.pyi:1391: note:     def __eq__(self, other: object) -> bool:
stubs/sublime.pyi:1391: note:         if not isinstance(other, Selection):
stubs/sublime.pyi:1391: note:             return NotImplemented
stubs/sublime.pyi:1391: note:         return <logic to compare two Selection instances>
```

Makes sense to me as comparison is not guaranteed to be done against the same type.

Reply:
Actually I put this on purpose to `Selection | None`.

The reason for that is because it might crash (i.e. raise a runtime exception) if you compare to an arbitrary object. The implementation in sulime.py is this:
```python
    def __eq__(self, rhs):
        return rhs is not None and list(self) == list(rhs)
``` 
So the argument is at least restricted to objects that are iterable. As you pasted above, Pyright indeed suggests a proper implementation which would make it work with any objects. But we don't have any influence on the implementation, so I would say here it's better to get a type checker warning with the current type stubs, instead of a potential runtime error in case of improper usage. What do you think?

Also the stub files are excluded from type checking in CI, so I guess errors/warnings for those files are not very important and we could ignore them.

Reply:
This is a mypy error. For `__lt__` it doesn't raise an issue.

But I don't fully understand what you mean by saying that the following implementation could crash:

```py
    def __eq__(self, rhs):
        return rhs is not None and list(self) == list(rhs)
```

if `rhs` is not a `Selection`. (Could be my ignorance regarding how `__eq__` works)

Reply:
```
Python 3.3>>> view.sel() == 1
Traceback (most recent call last):
  File "__main__", line 1, in <module>
  File "C:\Program Files\Sublime Text\Lib\python33\sublime.py", line 1094, in __eq__
    return rhs is not None and list(self) == list(rhs)
TypeError: 'int' object is not iterable
```
Similar for other classes which are not iterable.

If you run it in the 3.8 API it doesn't raise a runtime error because of the `isinstance` check:
```
>>> view.sel() == 1
False
```